### PR TITLE
add lightbox gallery and adjust site responsiveness for small screens

### PR DESCRIPTION
**Pull request by poncem91**
Commit adds multiple thumbnail pictures where the one picture was before and opens up a lightbox gallery when a thumbnail is clicked.
Also makes small changes to site in general to make it responsive to smaller screens and avoid collapsing the navbar.

### CODE REVIEW

File: portfolio/src/main/webapp/index.html [hunk 1]
```diff
@@ -98,5 +144,63 @@ <h1>Work Experience</h1>
...
     </div>
+
+    <!-- Lightbox Gallery -->
+          <div id="lightbox">
+              <a class="interface-buttons" id="close" onclick="closeLightbox();">Close X</a>
+              <div class="lightbox-content">
+                  <a class="interface-buttons" id="prev" onclick="changePic(-1)">&#10094;</a>
+                  <a class="interface-buttons" id="next" onclick="changePic(1)">&#10095;</a>
+                  <div class="lightbox-pic">
+                      <img src="/images/arlo1.jpg" alt="Arlo chilling in driveway">
```
Comment (peap) [inline portfolio/src/main/webapp/index.html:172]: You could also consider using `<figure>` and `<figcaption>` for these images: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/figcaption

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -64,4 +64,49 @@ function scrollToSection(sectionId) {
...
+function showPic(picIndex) {
+  const picsArray = document.getElementsByClassName("lightbox-pic");
+
+  if (picIndex > picsArray.length) {
+    picIndex = 1;	
+  }
+  
+  else if (picIndex < 1) {
+  	picIndex = picsArray.length;
+  }
```
Comment (peap) [inline portfolio/src/main/webapp/script.js]: Consider using the remainder operator (`%`) to keep your index within bounds here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Arithmetic_Operators#Remainder

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -98,5 +144,63 @@ <h1>Work Experience</h1>
           <h1>Contact Me</h1>
       </section>
     </div>
+
+    <!-- Lightbox Gallery -->
+          <div id="lightbox">
```
Comment (peap) [inline portfolio/src/main/webapp/index.html:166]: Totally optional, but if you want a little extra work, try making it possible to close the lightbox with the escape key, in addition to the close link.

**Review by peap — CHANGES_REQUESTED** [review]
Looks great!

File: portfolio/src/main/webapp/script.js (hunk 2 above)
Comment (poncem91) [reply portfolio/src/main/webapp/script.js]: This is a really clever way of wrapping around array index, thanks!!

File: portfolio/src/main/webapp/index.html (hunk 3 above)
Comment (poncem91) [reply portfolio/src/main/webapp/index.html:166]: Great idea! Added left and arrow key functionality as well while at it

File: portfolio/src/main/webapp/index.html (hunk 1 above)
Comment (poncem91) [reply portfolio/src/main/webapp/index.html:172]: Thanks!!

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -81,32 +82,37 @@ function openLightbox(picIndex) {
...
+
+  picsArray[picIndex].style.display = "block";
   currPic = picIndex;
+
 }
 
 // Previous and Next picture functionality
 function changePic(byIndex) {
     currPic += byIndex;
+    currPic = (currPic + picsArray.length) % picsArray.length;
```
Comment (peap) [inline portfolio/src/main/webapp/script.js:98]: Adding `picsArray.length` to `currPic` is redundant when taking the remainder of `picsArray.length`. (It's like multiplying by one or adding zero.) So you can just say `currPic %= picsArray.length`.

**Review by peap — APPROVED** [review]

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (poncem91) [reply portfolio/src/main/webapp/script.js:98]: I tried that at first, but when trying to chagePic from index 0 to 11, it computed -1 % 12 and outputted -1. I think % acts differently in JS, but let me know if I'm doing something wrong or if there's a better way to do this

Comment (peap) [reply portfolio/src/main/webapp/script.js:98]: Ah, yes, my bad. What you had originally is good; adding the length before taking the remainder is the way to go.

**Pull request merged**